src/Compilers/Core/Portable/Compilation.EmitStreamProvider.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// The compilation object is an immutable representation of a single invocation of the
+    /// compiler. Although immutable, a compilation is also on-demand, and will realize and cache
+    /// data as necessary. A compilation can produce a new compilation from existing compilation
+    /// with the application of small deltas. In many cases, it is more efficient than creating a
+    /// new compilation from scratch, as the new compilation can reuse information from the old
+    /// compilation.
+    /// </summary>
+    public abstract partial class Compilation
+    {
+        /// <summary>
+        /// Abstraction that allows the caller to delay the creation of the <see cref="Stream"/> values 
+        /// until they are actually needed.
+        /// </summary>
+        internal abstract class EmitStreamProvider
```

Comment:
@AlekseyTs I didn't change XML doc to this behavior because that would actually be a breaking change.  The XML file is rewritten on every compile, successful or not in the native compiler.  Hence I didn't change that behavior in my fix.  

I'm more than happy with making that change but I want to be very explicit about it.